src/DefaultBuilder/test/Microsoft.AspNetCore.Tests/WebApplicationTests.cs

```diff
@@ -165,32 +165,124 @@ public void WebApplicationBuilderWebHost_ThrowsWhenBuiltDirectly()
+            {
+                builder.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { WebHostDefaults.HostingStartupAssembliesKey, "hosting" }
+                });
+            }));
+
+            Assert.Throws<NotSupportedException>(() => builder.WebHost.ConfigureAppConfiguration(builder =>
+            {
+                builder.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { WebHostDefaults.HostingStartupExcludeAssembliesKey, "hostingexclude" }
+                });
+            }));
+        }
+
+        [Fact]
+        public void WebApplicationBuilderHostBuilderSettingsThatAffectTheHostCannotBeModified()
+        {
+            var builder = WebApplication.CreateBuilder();
+
+            var contentRoot = Path.GetTempPath().ToString();
+            var envName = $"{nameof(WebApplicationTests)}_ENV";
+
+            Assert.Throws<NotSupportedException>(() => builder.Host.ConfigureHostConfiguration(builder =>
+            {
+                builder.AddInMemoryCollection(new Dictionary<string, string>
+                {
+                    { HostDefaults.ApplicationKey, "myapp" }
+                });
+            }));
+            Assert.Throws<NotSupportedException>(() => builder.Host.UseEnvironment(envName));
+            Assert.Throws<NotSupportedException>(() => builder.Host.UseContentRoot(contentRoot));
+        }
+
+        [Fact]
+        public void WebApplicationBuilderCanModifyWebRootAfterCreateBuilder()
+        {
+            var builder = WebApplication.CreateBuilder();
+
+            var webRoot = Path.GetTempPath().ToString();
+
             builder.WebHost.UseSetting("WEBROOT", webRoot);
 
-            Assert.Equal(nameof(WebApplicationTests), builder.WebHost.GetSetting("APPLICATIONNAME"));
-            Assert.Equal(envName, builder.WebHost.GetSetting("environment"));
-            Assert.Equal(contentRoot, builder.WebHost.GetSetting("contentroot"));
             Assert.Equal(webRoot, builder.WebHost.GetSetting("webroot"));
 
             var app = builder.Build();
 
-            Assert.Equal(nameof(WebApplicationTests), app.Environment.ApplicationName);
-            Assert.Equal(envName, app.Environment.EnvironmentName);
-            Assert.Equal(contentRoot, app.Environment.ContentRootPath);
             Assert.Equal(webRoot, app.Environment.WebRootPath);
         }
 
+        [Fact]
+        public void WebApplicationBuilderWebRootIsRelativeToContentRoot()
+        {
+            var contentRoot = Path.GetTempPath().ToString();
+
+            var builder = WebApplication.CreateBuilder(new[] { $"--contentRoot={contentRoot}" });
+
+            builder.WebHost.UseSetting("WEBROOT", "wwwroot");
+
+            Assert.Equal("wwwroot", builder.WebHost.GetSetting("webroot"));
+
+            var app = builder.Build();
+
+            Assert.Equal(Path.Combine(contentRoot, "wwwroot"), app.Environment.WebRootPath);
```

Comment:
Do we still have any tests for a relative content root verifying it's relative to `AppContext.BaseDirectory`? It'd be similar to the first half of the `RelativePathsAreMappedToFullPaths`?

Reply:
I can add one.

Reply:
Will look into it.